core/src/main/java/org/elasticsearch/indices/memory/IndexingMemoryController.java

```diff
@@ -255,49 +250,45 @@ protected void updateShardBuffers(ShardId shardId, ByteSizeValue shardIndexingBu
         }
     }
 
+    /** returns {@code Boolean.TRUE} if no indexing operations have arrived to this shard since the {@code inactiveTimeNS},
+     *  {@code Boolean.FALSE} if they have, and {@code null} if the shard is unknown */
+    protected boolean isShardIdle(ShardId shardId, long inactiveTimeNS) {
+        final IndexShard shard = getShard(shardId);
+        if (shard == null) {
+            return false;
+        }
+        return currentTimeInNanos() - shard.getLastWriteNS() >= inactiveTimeNS;
```

Comment:
Shall we push this into IndexShard as well? have a method like

```
IndexShard.checkIsIdle() {
   if (System.currentTime - lastWriteNs() > inactiveTimeNs) {
         active.set(false);
  }
  return active.get() == false;
}
```

so all inactive/active logic is in IndexShard and you just ask for a check from the outside.